src/cpp/subscriber/Subscriber.cpp

```diff
@@ -49,7 +42,23 @@ bool Subscriber::init_middleware(const dds::xrce::OBJK_SUBSCRIBER_Representation
             break;
         }
     }
-    return rv;
+
+    return (created_entity ? new Subscriber(object_id, participant) : nullptr);
```

Comment:
We should avoid raw pointers. Maybe use smart pointers.

Reply:
Done!